src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala

```diff
@@ -29,13 +33,22 @@ class ConsignmentService(
   }
 
   def addConsignment(addConsignmentInput: AddConsignmentInput, userId: UUID): Future[Consignment] = {
-    val consignmentRow = ConsignmentRow(uuidSource.uuid, addConsignmentInput.seriesid, userId, Timestamp.from(timeSource.now))
-    consignmentRepository.addConsignment(consignmentRow).map(row => Consignment(row.consignmentid, row.userid, row.seriesid))
+    val transferStartYearForReference: Int = calendar.get(Calendar.YEAR)
+     consignmentRepository.getNextConsignmentSequence.flatMap(sequence => {
+       val consignmentRef = consignmentReferenceModel.createConsignmentReference(transferStartYearForReference, sequence)
+       val consignmentRow = ConsignmentRow(
+         uuidSource.uuid,
+         addConsignmentInput.seriesid,
+         userId,
+         Timestamp.from(timeSource.now),
+         consignmentsequence = Option(sequence))
+       consignmentRepository.addConsignment(consignmentRow).map(row => Consignment(row.consignmentid, row.userid, row.seriesid, Option(consignmentRef)))
```

Comment:
The function returns it so that the `consignmentService` function can then use the value when it adds the row to the Consignment table alongside the other Consignment row info. In the `consignmentService` file, line 39 & 46 (I don't know how to link to that line soz :sweat_smile: )